File: Programação 1/Exercicio 7/RMI/Cliente.py
import socket

def main():
    # Conectar ao SocketAdapter Java
    HOST = 'localhost'  # O endereço IP do servidor
    PORT = 5000         # A porta onde o SocketAdapter está escutando

    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.connect((HOST, PORT))

        # Leitura de dados do usuário
        idade = input("Digite a idade do funcionário: ")
        tempo_servico = input("Digite o tempo de serviço do funcionário: ")

        # Enviar dados para o servidor
        dados = f"{idade},{tempo_servico}\n"
        s.sendall(dados.encode())

        # Receber a resposta do servidor
        resposta = s.recv(1024).decode()
        print("Resposta do servidor:", resposta)

if __name__ == "__main__":
    main()

File: Programação 1/Exercicio 7/Sockets/Cliente.java
import java.io.*;
import java.net.*;
import java.util.Scanner;

public class Cliente {
    public static void main(String[] args) {
        String host = "localhost";
        int port = 5001;

        try (Socket socket = new Socket(host, port)) {
            // Conexão estabelecida

            // Criar input/output streams
            BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
            PrintWriter out = new PrintWriter(socket.getOutputStream(), true);
            
            try (// Leitura de dados do usuário
            Scanner scanner = new Scanner(System.in)) {
                System.out.print("Digite a idade do funcionário: ");
                int idade = scanner.nextInt();
                
                System.out.print("Digite o tempo de serviço do funcionário: ");
                int tempoServico = scanner.nextInt();

                // Enviar dados para o servidor
                out.println(idade + "," + tempoServico);
            }
            // Receber e exibir a resposta do servidor
            String resposta = in.readLine();
            System.out.println("Resposta do servidor: " + resposta);

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
